src/sharing/ShareWrapper.js

```diff
@@ -157,13 +154,37 @@ class ShareWrapperInner extends React.PureComponent<Props, State> {
       return;
     }
 
-    this.setSending();
+    this.setState({ sending: true });
     showToast(_('Sending message…'));
     if (sharedData.type === 'file') {
       const { files } = this.state;
       for (let i = 0; i < files.length; i++) {
-        const response = await api.uploadFile(auth, files[i].url, files[i].name);
-        messageToSend += `\n[${files[i].name}](${response.uri})\n`;
+        const { url, name: fileName } = files[i];
+        try {
+          const response = await api.uploadFile(auth, url, fileName);
+          messageToSend += `\n[${fileName}](${response.uri})\n`;
+        } catch (errorIllTyped) {
+          const error: mixed = errorIllTyped; // https://github.com/facebook/flow/issues/2470
+
+          if (!(error instanceof Error)) {
+            logging.error('ShareWrapper: Unexpected non-error thrown');
+          }
+
+          let msg = undefined;
+          if (
+            error instanceof ApiError
+            && error.code === 413 // 413 Payload Too Large:
+            //   https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/.2Fuser_uploads.3A.20.60413.20Payload.20Too.20Large.60/near/1332958
+          ) {
+            msg = _('This file is too large.');
```

Comment:
Let's make this say: "The server said the file is too large."

I.e., make clear that it's the server that made this judgement, not the client app.

In a lot of apps, where the server and client are provided by the same organization, making that distinction would be bad UI practice. But for Zulip I think it's appropriate because it helps the user figure out who to complain to.

(At first I wrote "The server rejected the file. The file is too large." (I considered "The server rejected the file as too large", but I think that syntax is a bit too complex for good UI writing.) But then while writing the next comment I realized that this gets shown below a heading saying "Failed to upload file". So it can be simplified.)